src/generator/UNetGANGenerator.py

```diff
@@ -70,7 +70,7 @@ def generate(self, latent_codes: np.ndarray) -> List[str]:
             FileNotFoundError: If any of the expected image output files were not found.
         """
         # Send codes via socket
-        em.send_latent_codes(latent_codes)
+        em.send_latent_codes(latent_codes.astype("float32"))
```

Comment:
wrong placement of astype in commit "fix: wrong type". Also use np.float32 instead of "float32"

Reply:
Once you are at it, maybe change "float32" to np.float32 in agent_asaddecoder.py as well!